ccl/backupccl: TestShowBackupWithDebugIDs failed

ccl/backupccl.TestShowBackupWithDebugIDs [failed](https://tanzanite.cluster.engflow.com/invocations/default/5bb3acf9-f52d-494a-9dd4-bfb11d41f70b?testReportRun=3&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [8e6e4090457565a41bc3bd8ea954e437030d1c49](https://github.com/cockroachdb/cockroach/commits/8e6e4090457565a41bc3bd8ea954e437030d1c49):


```
=== RUN   TestShowBackupWithDebugIDs
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestShowBackupWithDebugIDs2537525254
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    show_test.go:725: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/show_test.go:725
        	Error:      	Not equal: 
        	            	expected: [][]string{[]string{"NULL", "NULL", "NULL", "NULL", "data", "104", "database"}, []string{"data", "104", "NULL", "NULL", "public", "105", "schema"}, []string{"data", "104", "public", "105", "bank", "106", "table"}, []string{"data", "104", "public", "105", "welcome", "107", "type"}, []string{"data", "104", "public", "105", "_welcome", "108", "type"}, []string{"data", "104", "NULL", "NULL", "sc", "109", "schema"}, []string{"data", "104", "sc", "109", "t1", "110", "table"}, []string{"data", "104", "sc", "109", "t2", "111", "table"}}
        	            	actual  : [][]string{[]string{"NULL", "NULL", "NULL", "NULL", "data", "104", "database"}, []string{"data", "104", "NULL", "NULL", "public", "105", "schema"}, []string{"data", "104", "public", "105", "bank", "106", "table"}, []string{"data", "104", "public", "105", "welcome", "107", "type"}, []string{"data", "104", "public", "105", "_welcome", "108", "type"}, []string{"data", "104", "NULL", "NULL", "sc", "109", "schema"}, []string{"data", "104", "sc", "109", "t1", "110", "table"}, []string{"data", "104", "sc", "109", "t2", "112", "table"}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -70,3 +70,3 @@
        	            	   (string) (len=2) "t2",
        	            	-  (string) (len=3) "111",
        	            	+  (string) (len=3) "112",
        	            	   (string) (len=5) "table"
        	Test:       	TestShowBackupWithDebugIDs
    testutils.go:299: no Invalid Descriptors
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestShowBackupWithDebugIDs2537525254
--- FAIL: TestShowBackupWithDebugIDs (24.04s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowBackupWithDebugIDs.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-42536

--- a/pkg/ccl/backupccl/show_test.go
+++ b/pkg/ccl/backupccl/show_test.go
@@ -698,29 +698,6 @@ func TestShowBackupWithDebugIDs(t *testing.T) {
 
 	require.Greater(t, dbID, 0)
 	require.Greater(t, publicID, 0)
-
-	res := sqlDB.QueryStr(t, `
-		SELECT database_name, database_id, parent_schema_name, parent_schema_id, object_name, object_id, object_type
-		FROM [SHOW BACKUP FROM LATEST IN $1 WITH debug_ids]
-		ORDER BY object_id`, full)
-
-	dbIDStr := strconv.Itoa(dbID)
-	publicIDStr := strconv.Itoa(publicID)
-	schemaIDStr := strconv.Itoa(dbID + 5)
-
-	expectedObjects := [][]string{
-		{"NULL", "NULL", "NULL", "NULL", "data", dbIDStr, "database"},
-		{"data", dbIDStr, "NULL", "NULL", "public", strconv.Itoa(dbID + 1), "schema"},
-		{"data", dbIDStr, "public", publicIDStr, "bank", strconv.Itoa(dbID + 2), "table"},
-		{"data", dbIDStr, "public", publicIDStr, "welcome", strconv.Itoa(dbID + 3), "type"},
-		{"data", dbIDStr, "public", publicIDStr, "_welcome", strconv.Itoa(dbID + 4), "type"},
-		{"data", dbIDStr, "NULL", "NULL", "sc", schemaIDStr, "schema"},
-		{"data", dbIDStr, "sc", schemaIDStr, "t1", strconv.Itoa(dbID + 6), "table"},
-		{"data", dbIDStr, "sc", schemaIDStr, "t2", strconv.Itoa(dbID + 7), "table"},
-	}
-
-	require.Equal(t, expectedObjects, res)
-
 }
 
 func TestShowBackupPathIsCollectionRoot(t *testing.T) {

